consensus/src/common.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use blake2_rfc::blake2b::blake2b;
+use mina_crypto::hash::BaseHash;
+use mina_rs_base::consensus_state::ConsensusState;
+use mina_rs_base::global_slot::GlobalSlot;
+use mina_rs_base::protocol_state::{Header, ProtocolState};
+use serde_bin_prot::to_writer;
+use std::convert::TryInto;
+
+pub struct ProtocolStateChain(Vec<ProtocolState>);
+
+pub trait Chain<T>
+where
+    T: Header,
+{
+    fn push(&mut self, new: T) -> Result<(), &'static str>;
```

Comment:
The second type param of Result needs to be a proper error type. We can define and enum ConsensusError which derives thiserror::Error from https://docs.rs/thiserror/1.0.26/thiserror/ crate use it in place of &'static str

Reply:
Seconded, check out https://github.com/ChainSafe/serde-bin-prot/blob/main/src/error.rs for an example of writing a custom error type

Reply:
updated!